### PR TITLE
LLVM 14.0.3

### PR DESCRIPTION
**Pull request by h-vetinari**
Manual PR because the bot doesn't work anymore now that this recipe has several LLVM versions at the same time.

Blockers for merging this PR and thus enabling the compilers in conda-forge (indentation denotes dependency; c.f. list from [13.0.1](https://github.com/conda-forge/clang-compiler-activation-feedstock/pull/77#issuecomment-1030635626)):

* [x] https://github.com/conda-forge/llvmdev-feedstock/pull/152
  * [x] https://github.com/conda-forge/clangdev-feedstock/pull/170
    * [x] https://github.com/conda-forge/cctools-and-ld64-feedstock/pull/56
    * [x] https://github.com/conda-forge/compiler-rt-feedstock/pull/39
    * [x] https://github.com/conda-forge/libcxx-feedstock/pull/85
  * [x] https://github.com/conda-forge/lld-feedstock/pull/37
  * [x] https://github.com/conda-forge/openmp-feedstock/pull/62
 
Additionally in the LLVM-space:
* [ ] https://github.com/conda-forge/mlir-feedstock/pull/14
* [x] https://github.com/conda-forge/clang-win-activation-feedstock/pull/15

### CODE REVIEW

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -34,3 +34,4 @@ zip_keys:
 version:
   - 12.0.1
   - 13.0.1
+  - 14.0.0
```
Comment (h-vetinari) [inline recipe/conda_build_config.yaml:37]:
```suggestion
  - 14.0.3
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,9 +1,9 @@
 {% if version is not defined %}
-{% set version = "13.0.1" %}
+{% set version = "14.0.0" %}
```
Comment (h-vetinari) [inline recipe/meta.yaml:2]:
```suggestion
{% set version = "14.0.3" %}
```

**Comment by h-vetinari** [comment]
@conda-forge-admin, please rerender

**Comment by h-vetinari** [comment]
@isuruf, this should be the final puzzle piece for LLVM 14. It's a pretty trivial update so I'll merge in 2-3 days if there are no other comments, but wanted to double-check with you nevertheless.

**Comment by isuruf** [comment]
Thanks

**Pull request merged**